components/omega/doc/design/VerticalMixingCoeff.md

```diff
@@ -0,0 +1,182 @@
+3. `ConvectiveTriggerBVF`. Value of Brunt Viasala frequency squared below which convective mixing is triggered. \[0.0\]
+
+The following config options should be included for background vertical mixing:
+
+1. `BackgroundViscosity`. Background vertical viscosity applied to velocity components. \[1.0e-4 $m^2 s^{-1}$\]
+2. `BackgroundDiffusivity`. Background vertical diffusivity applied to all tracer quantities. \[1.0e-5 $m^2 s^{-1}$\]
+
+### 4.2 Methods
+
+<!--- List and describe all public methods and their interfaces (actual code for
+interface that would be in header file). Describe typical use cases. -->
+
+It is assumed that viscosity and diffusivity will be stored at cell centers. Additionally, since they will be used elsewhere, it is assumed that zMid and N2 have been computed elsewhere (N2 in the density routine) and stored. Quantities such as squared shear and Richardson number are not be needed outside of the vertical mixing routines, so are computed locally.
+
+```c++
+parallelFor(
+    {NCellsAll, NVertLevels}, KOKKOS_LAMBDA(int ICell, int K) {
+
+        VertViscosity(ICell, K) = 0._Real;
+        VertDiffusivity(ICell, K) = 0._Real;
+
+        // If shear mixing true, add shear contribution to viscosity and diffusivity 
+        if (EnableShearMix) {
+            const Real InvAreaCell = 1._Real / AreaCell(ICell);
+            // Calculate the square of the shear
+            for (int J = 0; J < NEdgesOnCell(ICell); ++J) {
+                const I4 JEdge = EdgesOnCell(ICell, J);
+                const Real Factor = 0.5_Real * DcEdge(JEdge) * DvEdge(JEdge) * InvAreaCell;
+                const Real DelU2 = pow(NormalVelocity(JEdge,K-1) - NormalVelocity(JEdge,K), 2._Real) + pow(TangentialVelocity(JEdge,K-1) - TangentialVelocity(JEdge,K), 2._Real);
+                ShearSquared(ICell, K) = ShearSquared(ICell, K) + Factor * DelU2;
+            }
+            ShearSquared(ICell, K) = ShearSquared(ICell, K) / pow(zMid(ICell, K-1) - zMid(ICell, K), 2._Real);
+
+            // Calculate Richardson number
+            RichardsonNum(ICell, K) = BruntVaisalaFreq(ICell, K) / max(ShearSquared(ICell, K), 1.0e-12_Real);
+
+            // Add in shear contribution to vertical mixing
+            VertViscosity(ICell, K) = VertViscosity(ICell, K) + ShearNuZero / pow(1._Real + ShearAlpha * RichardsonNum(ICell, K), ShearExponent);
+            VertDiffusivity(ICell, K) = VertDiffusivity(ICell, K) + VertViscosity(ICell, K) / (1 + ShearAlpha * RichardsonNum(ICell, K));
+        }
+
+        // If conv mixing true, add conv contribution to viscosity and diffusivity 
+        if (EnableConvectiveMix) {
+            if (BruntVaisalaFreq(ICell, K) < ConvectiveTriggerBVF) {
+                VertViscosity(ICell, K) = VertViscosity(ICell, K) + ConvectiveDiffusivity;
+                VertDiffusivity(ICell, K) = VertDiffusivity(ICell, K) + ConvectiveDiffusivity;
+            }
+        }
+
+        // Add background contribution to viscosity and diffusivity 
```

Comment:
I would actually move this to the top and use the background values to initialize, rather than initializing to zero and adding these later. That will save you an addition and some memory ops.

Reply:
Done!